src/bigint.rs

```diff
@@ -2707,7 +2707,9 @@ impl TryFrom<&BigInt> for BigUint {
 
     #[inline]
     fn try_from(value: &BigInt) -> Result<BigUint, TryFromBigIntError<()>> {
-        value.to_biguint().ok_or(TryFromBigIntError::new(()))
+        value
+            .to_biguint()
+            .ok_or_else(|| TryFromBigIntError::new(()))
```

Comment:
I think clippy is being overly fussy here since `TryFromBigIntError<()>` is a ZST, and `new` isn't doing anything significant, but I guess there's no harm in wrapping this. Another option would be to write this as a direct struct initialization for the ZST, but I think it won't matter for performance either way.

Reply:
Ah. Yeah, I think it's probably the cleanest way to shut Clippy up.